labelbox/schema/project.py

```diff
@@ -1260,38 +1270,47 @@ def _is_url_valid(url: Union[str, Path]) -> bool:
             raise ValueError(
                 f'Invalid annotations given of type: {type(annotations)}')
 
-    def _wait_until_data_rows_are_processed(self,
-                                            data_row_ids: List[str],
-                                            wait_processing_max_seconds: int,
-                                            sleep_interval=30):
+    def _wait_until_data_rows_are_processed(
+            self,
+            data_row_ids: List[str] = None,
+            global_keys: List[str] = None,
+            wait_processing_max_seconds: int = _wait_processing_max_seconds,
+            sleep_interval=30):
         """ Wait until all the specified data rows are processed"""
         start_time = datetime.now()
+
         while True:
             if (datetime.now() -
                     start_time).total_seconds() >= wait_processing_max_seconds:
                 raise ProcessingWaitTimeout(
                     "Maximum wait time exceeded while waiting for data rows to be processed. Try creating a batch a bit later"
                 )
 
-            all_good = self.__check_data_rows_have_been_processed(data_row_ids)
+            all_good = self.__check_data_rows_have_been_processed(
+                data_row_ids, global_keys)
             if all_good:
                 return
 
             logger.debug(
                 'Some of the data rows are still being processed, waiting...')
             time.sleep(sleep_interval)
 
-    def __check_data_rows_have_been_processed(self, data_row_ids: List[str]):
+    def __check_data_rows_have_been_processed(self, data_row_ids: List[str],
+                                              global_keys: List[str]):
         data_row_ids_param = "data_row_ids"
+        global_keys_param = "global_keys"
 
-        query_str = """query CheckAllDataRowsHaveBeenProcessedPyApi($%s: [ID!]!) {
-            queryAllDataRowsHaveBeenProcessed(dataRowIds:$%s) {
+        query_str = """query CheckAllDataRowsHaveBeenProcessedPyApi($%s: [ID!], $%s: [ID!]) {
+            queryAllDataRowsHaveBeenProcessed(dataRowIds:$%s, globalKeys:$%s) {
                 allDataRowsHaveBeenProcessed
            }
-        }""" % (data_row_ids_param, data_row_ids_param)
+        }""" % (data_row_ids_param, global_keys_param, data_row_ids_param,
+                global_keys_param)
 
-        params = {}
-        params[data_row_ids_param] = data_row_ids
+        params = {
```

Comment:
do we have to send both if only one will be ever present?

Reply:
I'll see if I can pass just one.

Reply:
Updated so that only one of globalKeys or dataRowIds is passed.